### DISCUSSION TITLE
Autocompleting the `in` from `let..in` into `in_channel_length`: can this be prevented?

### DISCUSSION THREAD

**Discussion by ELLIOTTCABLE** [ocamllabs/vscode-ocaml-platform#955] [Q&A]
Constantly, absolutely constantly, while writing, I get popups like this:

<img width="734" alt="Screen Shot 2022-05-20 at 11 20 06 AM" src="https://user-images.githubusercontent.com/200/169570732-adc2ef59-9e1f-491f-b0c5-b876a5606a7f.png">

It's gotten to the point where my personal muscle-memory for "I want to go to a new line" has migrated from <kbd>Return</kbd> to <kbd>ESC</kbd><kbd>Return</kbd> instead, because I'm always expecting an unwanted `in_channel_length`.

Is there a way to:

1. Add `in` itself as a completion, so <kbd>Return</kbd> selects it first;
2. prevent completions at the end of `let..in` constructs entirely;
3. or otherwise fix this behaviour?

Relatedly, is this something worth opening a bug-report for — could it be pre-emptively fixed for all VSCode OCaml Platform users?

**Comment by mlantas** [comment]
This was fixed at one point, but I think there were problems that caused it to be removed unfortunately.

- https://github.com/ocamllabs/vscode-ocaml-platform/issues/183
- https://github.com/ocaml/ocaml-lsp/issues/173
- https://github.com/ocaml/ocaml-lsp/issues/420

**Reply by OscarLahaie** [reply]
Is there a way to fix it or does something prevent it?
Edit: Can we overcome the proposal made by OCaml-lsp?

**Reply by OscarLahaie** [reply]
Maybe we could make some keywords available from the extension (those unavailable in lsp) and call the rest from ocaml-lsp.

**Reply by rgrinberg** [reply]
We'd cause the opposite problem: keyword completions in places where keywords couldn't possibly be used, e.g. completing identifiers.
